pkg/kontext/bundle.go

```diff
@@ -47,19 +47,49 @@ func bundle(directory string) (v1.Layer, error) {
 	buf := bytes.NewBuffer(nil)
 	tw := tar.NewWriter(buf)
 	defer tw.Close()
-
+	var excludedDirs []string = make([]string, 0)
 	err := filepath.Walk(directory,
 		func(path string, fi os.FileInfo, err error) error {
 			if err != nil {
 				return err
 			}
 
 			// Skip anything in the .git directory
-			// TODO(mattmoor): expand this to .gitignore / .dockerignore?
+
 			if fi.IsDir() && filepath.Base(path) == ".git" {
 				return filepath.SkipDir
 			}
 
+			//Check if the directory has  .dockerignore
+			hasDockerIgnore, fDockerIgnore, err := hasDockerIgnore(directory)
+
+			if err != nil {
+				return err
+			}
+
+			var patterns []string = make([]string, 0)
+			if hasDockerIgnore {
```

Comment:
As I allude to elsewhere, I think we should define an "ignorer" of sorts, and just always have one.

This can either be constructed from a `.dockerignore` or be some built-in heuristics (like `.git/` above).

Reply:
I have done away with it in the new refactor